carbon-design-system/carbon
# feat(Tabs): Contained tabs on the grid

Closes #13353 

Adds `fullWidth` prop to `TabList` so that Tabs (container) can be aligned to the Grid

#### Changelog

**New**

- `fullWidth` prop now available for contained `TabList`
- 'contained fullWidth' tab story and e2e test
- tests for `fullWidth` prop and added mock for `useMatchMedia` in Tab testing
- 'cds-tabs--full-width' class for `fullWidth` contained Tabs and accompanying style
- Grid guidance using `fullWidth` prop  for Tabs


#### Testing / Reviewing

1. Check that stories match spec: https://github.com/carbon-design-system/carbon/issues/13351#issuecomment-1500455606
     - https://deploy-preview-13927--carbon-components-react.netlify.app/?path=/story/components-tabs--contained-full-width 
     - https://deploy-preview-13927--carbon-components-react.netlify.app/iframe.html?args=&id=components-tabs--contained-grid-test&viewMode=story 
3. Check developer guidance is true and coherent: https://deploy-preview-13927--carbon-components-react.netlify.app/?path=/docs/components-tabs--contained-full-width#tabs-and-the-grid---fullwidth-prop
4. Should have no visual/behavioral regression to default tab styles and functionality outside of `fullWidth`


TODO:

- [x] Remove Test story
